src/test/java/com/purbon/kafka/topology/actions/topics/builders/TopicConfigUpdatePlanBuilderTest.java

```diff
@@ -0,0 +1,125 @@
+package com.purbon.kafka.topology.actions.topics.builders;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doReturn;
+
+import com.purbon.kafka.topology.TopicManager;
+import com.purbon.kafka.topology.actions.topics.TopicConfigUpdatePlan;
+import com.purbon.kafka.topology.api.adminclient.TopologyBuilderAdminClient;
+import com.purbon.kafka.topology.model.Impl.TopicImpl;
+import com.purbon.kafka.topology.model.Topic;
+import java.util.Collections;
+import java.util.HashMap;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.common.config.TopicConfig;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class TopicConfigUpdatePlanBuilderTest {
+
+  private static final String TOPIC_NAME = "foo";
+  public static final String DEFAULT_RETENTION_MS = "604800000";
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Mock private TopologyBuilderAdminClient adminClient;
+
+  @Test
+  public void shouldNotChangeConfigWhenNoConfig() {
+    doReturn(createEmptyConfig()).when(adminClient).getActualTopicConfig(TOPIC_NAME);
+    var topic = createTopic();
+    var plan = getTopicConfigUpdatePlan(topic);
+    assertNewUpdatedAndDeletedCounts(plan, 0, 0, 0);
+  }
+
+  @Test
+  public void shouldNotAddNewConfigForNumPartitionsButShouldUpdateFlag() {
+    doReturn(createEmptyConfig()).when(adminClient).getActualTopicConfig(TOPIC_NAME);
+    var topic = createTopic(TopicManager.NUM_PARTITIONS, "5");
+    var plan = getTopicConfigUpdatePlan(topic);
+    assertNewUpdatedAndDeletedCounts(plan, 0, 0, 0);
+    assertTrue(plan.isUpdatePartitionCount());
+  }
+
+  @Test
+  public void shouldAddNewConfigForRetention() {
+    doReturn(createDefaultRetentionConfig()).when(adminClient).getActualTopicConfig(TOPIC_NAME);
+    var topic = createTopic(TopicConfig.RETENTION_MS_CONFIG, "1000");
+    var plan = getTopicConfigUpdatePlan(topic);
+    assertNewUpdatedAndDeletedCounts(plan, 1, 0, 0);
+  }
+
+  @Test
+  public void shouldUpdateConfigForRetention() {
+    doReturn(createAlreadyOverriddenRetentionConfig())
+        .when(adminClient)
+        .getActualTopicConfig(TOPIC_NAME);
+    var topic = createTopic(TopicConfig.RETENTION_MS_CONFIG, "1000");
+    var plan = getTopicConfigUpdatePlan(topic);
+    assertNewUpdatedAndDeletedCounts(plan, 0, 1, 0);
+  }
+
+  @Test
+  public void shouldDeleteConfigForRetention() {
```

Comment:
Delete for forskjellege configSources?

Reply:
Siden vi nå _kun_ sjekker på DYNAMIC_TOPIC_CONFIG i koden, holder det å teste med én annen configSource. Har lagt til testen `shouldNotDeleteConfigForRetentionWhenSetByBroker`.